sql/src/test/java/io/crate/integrationtests/CircuitBreakerIntegrationTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import io.crate.breaker.CrateCircuitBreakerService;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1, supportsDedicatedMasters = false, numClientNodes = 0)
+public class CircuitBreakerIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Test
+    public void testQueryBreakerIsDecrementedWhenQueryCompletes() {
+        execute("create table t1 (text string)");
+
+        StringBuilder longText = new StringBuilder();
+        for (int i = 0; i < 8_200; i++) {
+            longText.append(String.valueOf(i));
+        }
+
+        // insert ~3mb of data so that the data we accumulate in the RamAccountingContexts will be flushed into the
+        // underlying ES circuit breaker (on which we'll assert the data was unaccounted for after the query completes)
+        for (int i = 0; i < 100; i++) {
+            execute("insert into t1 values ('" + i + longText.toString() + "')");
```

Comment:
use parameter substitution? I know, irrelevant for tests - but inlining values with string formatting shouldn't become a habit. (Ideally also use a single bulk requests to have a faster test)